sdk/deepbook-v3/src/client.ts

```diff
@@ -0,0 +1,433 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import { bcs } from '@mysten/sui/bcs';
+import type { SuiClient } from '@mysten/sui/client';
+import { Transaction } from '@mysten/sui/transactions';
+import { normalizeSuiAddress } from '@mysten/sui/utils';
+
+import { BalanceManagerContract } from './transactions/balanceManager.js';
+import { DeepBookContract } from './transactions/deepbook.js';
+import { DeepBookAdminContract } from './transactions/deepbookAdmin.js';
+import { FlashLoanContract } from './transactions/flashLoans.js';
+import { GovernanceContract } from './transactions/governance.js';
+import type { BalanceManager, Environment } from './types/index.js';
+import { DEEP_SCALAR, DeepBookConfig, FLOAT_SCALAR } from './utils/config.js';
+import type { CoinMap, PoolMap } from './utils/constants.js';
+
+/**
+ * DeepBookClient class for managing DeepBook operations.
+ */
+export class DeepBookClient {
+	client: SuiClient;
+	#config: DeepBookConfig;
+	#address: string;
+	balanceManager: BalanceManagerContract;
+	deepBook: DeepBookContract;
+	deepBookAdmin: DeepBookAdminContract;
+	flashLoans: FlashLoanContract;
+	governance: GovernanceContract;
+
+	/**
+	 * @param {SuiClient} client SuiClient instance
+	 * @param {string} address Address of the client
+	 * @param {Environment} env Environment configuration
+	 * @param {Object.<string, BalanceManager>} [balanceManagers] Optional initial BalanceManager map
+	 * @param {CoinMap} [coins] Optional initial CoinMap
+	 * @param {PoolMap} [pools] Optional initial PoolMap
+	 * @param {string} [adminCap] Optional admin capability
+	 */
+	constructor({
+		client,
+		address,
+		env,
+		balanceManagers,
+		coins,
+		pools,
+		adminCap,
+	}: {
+		client: SuiClient;
+		address: string;
+		env: Environment;
+		balanceManagers?: { [key: string]: BalanceManager };
+		coins?: CoinMap;
+		pools?: PoolMap;
+		adminCap?: string;
+	}) {
+		this.client = client;
+		this.#address = normalizeSuiAddress(address);
+		this.#config = new DeepBookConfig({
+			address: this.#address,
+			env,
+			balanceManagers,
+			coins,
+			pools,
+			adminCap,
+		});
+		this.balanceManager = new BalanceManagerContract(this.#config);
+		this.deepBook = new DeepBookContract(this.#config);
+		this.deepBookAdmin = new DeepBookAdminContract(this.#config);
+		this.flashLoans = new FlashLoanContract(this.#config);
+		this.governance = new GovernanceContract(this.#config);
+	}
+
+	setConfig(config: DeepBookConfig) {
```

Comment:
do we need this, I only see it in one test.

It looks like there might not currently be a way to construct the client with certain config values.  I think I'd rather update the constructor to optionally accept a config instance instead.  It's probably better to create a new client when the config changes